### PR TITLE
Update the releasing doc

### PR DESCRIPTION
**Pull request by charlesbvll**
### Description

The release process was simplified in a recent PR but the docs don't reflect the change.

### Related issues/PRs

N/A

### Explanation

Describe the new process in the docs.

### Checklist

- [x] Implement proposed change
- [x] Make CI checks pass
- [x] Ping maintainers on [Slack](https://flower.dev/join-slack/) (channel `#contributions`)

### Any other comments?

N/A

### CODE REVIEW

File: doc/source/contributor-how-to-release-flower.rst [hunk 1]
```diff
@@ -20,8 +19,8 @@ The version number of a release is stated in ``pyproject.toml``. To release a ne
 
 1. Update the ``changelog.md`` section header ``Unreleased`` to contain the version number and date for the release you are building. Create a pull request with the change.
 2. Tag the release commit with the version number as soon as the PR is merged: ``git tag v0.12.3``, then ``git push --tags``
-3. Build the release with ``./dev/build.sh``, then publish it with ``./dev/publish.sh``
-4. Create an entry in GitHub releases with the release notes for the previously tagged commit and attach the build artifacts (:code:`.whl` and :code:`.tar.gz`).
+
+A draft release will automatically be created on GitHub with the correct artifacts, verify that everything looks good on it, and, if so, publish it.
```
Comment (danieljanes) [inline doc/source/contributor-how-to-release-flower.rst:23]: Let's reword this as `3. ...` and add the note about the draft release that gets created to the previous point `2.` so that it's easy to understand which action causes what effect

**Review by danieljanes — CHANGES_REQUESTED** [review]

File: doc/source/contributor-how-to-release-flower.rst (hunk 1 above)
Comment (charlesbvll) [reply doc/source/contributor-how-to-release-flower.rst]: @danieljanes Just changed it, wdyt?

**Review by danieljanes — APPROVED** [review]
